pkg/catalogv2/system/system.go

```diff
@@ -40,6 +39,13 @@ var (
 			"system:masters",
 		},
 	}
+
+	// noVersionRefresh maps charts which should be updated conditionally when Rancher restarts, to version settings
+	// corresponding to those charts.
+	noVersionRefresh = map[string]settings.Setting{
```

Comment:
Thanks, suggestion applied, ensuring that a min version is treated as an exact version if non-empty.
I think we may need further tests to ensure that manual Fleet updates triggered via the web UI would not be undone by catalog refreshes (even without Fleet-specific changes implemented via this PR, see `installCharts`).

Reply:
Confirmed, removing [this block](https://github.com/rancher/rancher/pull/42304/files#diff-96f7551d1bcafc092b3502857d33f73f576c753e3a4311b8660fabc446b383b0R207) leads to a manual update of Fleet (tried from 0.7.1 to 0.9.0-rc.1) being automatically undone when the catalog is refreshed.

Reply:
When you performed the above check, did you also remove the changes [here](https://github.com/rancher/rancher/pull/42304/files#diff-20d8d49e9283f7a7a9df8a56f972c1f228ea6f019c4908e4a670a9f020e7c16aL79-R93)? If you pass in both exact and minVersion you will not see the desired results. Since my suggested change is dependent on the exact version being empty.

 > Fleet being the only charts using a min version setting, perhaps we could get rid of min version support altogether (eg. via another PR).
 
If we add the suggested changes above this PR will be changing the behavior of minVersion, so it would be acceptable to change the name from `minVersion` to something more appropriate like `defaultVersion`.

Reply:
>  If you pass in both exact and minVersion you will not see the desired results.
I've repeated the above check without the mentioned changes, ie passing a non-empty min version and an empty exact version, with the same results: without version override logic in the system charts controller, Fleet is downgraded when the catalog is refreshed.